Feature/46 feature 프로젝트 삭제

a121255 opened the pull request:
# 🚀 Pull Request

프로젝트 삭제 기능 추가

## #️⃣ 연관된 이슈

#46 

## 📋 작업 내용

- Project
  - 삭제여부는 deleted_yn으로만 관리하도록 수정
  - 엔티티의 일부 값만을 변경할 수 있도록 toBuilder = true 옵션 추가
- ProjectRequest
  - 프로젝트 생성 시 deleted_yn의 기본값은 'N'이 되도록 수정
- ProjectResponse
  - response에서 삭제여부 값을 확인할 수 있도록 수정
- ProjectController
  - 기존 URI 수정 및 삭제 메서드 추가
- ProjectService, ProjectServiceImpl
  - CustomException이 deprecated 되어 BaseException으로 수정
  - 삭제 메서드 추가 (soft delete)

Commit:
feat : 프로젝트 entity, dto 수정

- Project
  - 삭제여부는 deleted_yn으로만 관리하도록 수정
  - 엔티티의 일부 값만을 변경할 수 있도록 toBuilder = true 옵션 추가
- ProjectRequest
  - 프로젝트 생성 시 deleted_yn의 기본값은 'N'이 되도록 수정
- ProjectResponse
  - response에서 삭제여부 값을 확인할 수 있도록 수정

Commit:
feat : 프로젝트 service, controller 수정

- ProjectController
  - 기존 URI 수정 및 삭제 메서드 추가
- ProjectService, ProjectServiceImpl
  - CustomException이 deprecated 되어 BaseException으로 수정
  - 삭제 메서드 추가 (soft delete)

Commit:
test : 프로젝트 삭제 테스트 추가

- 프로젝트 삭제 테스트 추가

oowtl commented on module-service/src/main/java/com/checkping/service/project/ProjectServiceImpl.java:
```diff
@@ -22,9 +24,25 @@ public class ProjectServiceImpl implements ProjectService {
     @Override
     public ProjectResponse.ProjectDto registerProject(ProjectRequest.ResisterDto request) {
         if(request == null) {
```
코드의 내용을 보아 controller 부분에서 `@RequestBody(required=true)` 도 좋은 방법이 될 것 같아요~

Commit:
feat : 파라미터 유효성 검사 수정

- ProjectServiceImpl
  - required는 생략할 경우 true가 default, 해당 옵션이 null 체크를 진행해주기 때문에 기존의 if문을 제거
  - ,사용자 입력값 중 필수값이 name 정도로 생각되어 해당 값에 대한 유효성 검사를 추가
